linq.go

```diff
@@ -1218,6 +1219,42 @@ func (q Query) Average() (avg float64, err error) {
 	return
 }
 
+// Zip returns a query with the result of the "zipping" function
+// provided
+//
+// Example:
+// convertRoman:= func(n, m T) (T, error){
+// return n.(string) + " - " + m.(string), nil
+// }
+//	result, err := From([]string {"i", "ii", "iii", "iv", "v"})
+// .Zip([]string {"one", "two", "three", "four", "five"}, convertRoman)
+// .Results()
+func (q Query) Zip(second T, f func(T, T) (T, error)) (r Query) {
+	if q.err != nil {
+		r.err = q.err
+		return r
+	}
+	if f == nil {
+		r.err = ErrNilFunc
+		return
+	}
+	q2 := From(second)
+	if q2.err != nil {
+		r.err = q2.err
+		return r
+	}
+	maxIterationCount := int (math.Min(float64(len(q.values)), float64(len(q2.values))))
```

Comment:
Also you can use `max=b; if a>b{max=a}`. No need to convert to float and use math.floor here